### PR TITLE
Added support for modified sum and difference loss from https://arxiv.org/abs/2208.11428

### PR DESCRIPTION
**Pull request by sai-soum**
image

### CODE REVIEW

File: auraloss/freq.py [hunk 1]
```diff
@@ -604,3 +604,175 @@ def forward(self, input: torch.Tensor, target: torch.Tensor):
             return loss
         elif self.output == "full":
             return loss, sum_loss, diff_loss
+
+class ModifiedSumAndDifferenceSTFTLoss(torch.nn.Module):
```
Comment (csteinmetz1) [inline auraloss/freq.py:608]: Instead of creating a new class for this we should add parameters to `SumAndDifferenceSTFTLoss` in order to support this behavior. It seems like the major modification is that application of the pre-emphasis filter.

File: auraloss/freq.py [hunk 2]
```diff
@@ -604,3 +604,175 @@ def forward(self, input: torch.Tensor, target: torch.Tensor):
...
+        target_sum_mag, target_sum_phs = self.stft(target_sum.view(-1, target_sum.size(-1)))
+        target_diff_mag, target_diff_phs = self.stft(target_diff.view(-1, target_diff.size(-1)))
+
+        # compute loss terms
+       
+        #compute the L1 log magnitude loss
+        l1log_sum_loss = self.l1logstft(input_sum_mag, target_sum_mag)
+        l1log_diff_loss = self.l1logstft(input_diff_mag, target_diff_mag)
+
+        if self.loss_type == 'SClogL1':
```
Comment (csteinmetz1) [inline auraloss/freq.py:749]: Looks like the other difference in the distance measure. This should be able to be supported by the main class. However, if it seems easier, we could consider adding a new `ModifiedSumAndDifferenceLoss` class but where it inherits from the main class so that we don't get all this repeated code (e.g. `stft`, etc.)

**Review by csteinmetz1 — CHANGES_REQUESTED** [review]
Thanks, this will be nice to add. Main point is that we should avoid creating a new class and extend the existing class to support this behavior if possible. If that doesn't workout, at least make the new class inherit from the main class to avoid any repetitive code.
